### PR TITLE
Temporarily remove LanguageClient dependency

### LINKED ISSUE #17: ✨ Temporarily Remove LanguageClient Dependency
### Description

In the main CodeEdit repo, the CodeEditKit library's dependencies are preventing me from including my LSP changes. I've removed the dependency and commented out the current LSP file so that we can start running LSP on the main repo.

### Alternatives Considered

I've considered just upgrading the dependency but I figured keeping the LanguageClient dependency in the main repo will allow me to iterate on the LSP code quicker.

### Additional Context

_No response_

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by FastestMolasses**
### Description

Removes the LanguageClient and ProcessService dependency from this library so that it can be added and updated within the main CodeEdit repo instead. This is just temporary to allow the LSP code to exist and be updated without relying on CodeEditKit's version.

### Related Issues

Closes #17 

### Checklist

- [X] I read and understood the [contributing guide](https://github.com/CodeEditApp/CodeEdit/blob/main/CONTRIBUTING.md) as well as the [code of conduct](https://github.com/CodeEditApp/CodeEdit/blob/main/CODE_OF_CONDUCT.md)
- [X] The issues this PR addresses are related to each other
- [X] My changes generate no new warnings
- [X] My code builds and runs on my machine
- [X] My changes are all related to the related issue above
- [X] I documented my code

### CODE REVIEW

**Comment by austincondiff** [comment]
SwiftLint is failing, is this temporary too or can you fix this?

File: Sources/CodeEditKit/NonUIExtensions/LSP/LanguageExtension.swift [hunk 1]
```diff
@@ -5,146 +5,146 @@
...
+//            self.wrappedServer.sendRequest(request, completionHandler: { (result: ServerResult<Response>) in
+//                if case .success = result, case .shutdown = request {
+//                    self.stopProcess()
+//                }
+//
+//                completionHandler(result)
+//            })
+//        }
+//    }
+//}
```
Comment (activcoding) [inline Sources/CodeEditKit/NonUIExtensions/LSP/LanguageExtension.swift:150]: This should fix the swift lint warnings:
```suggestion
// import Foundation
// import LanguageClient
// import Combine
// import OSLog
// import ProcessServiceClient
// import ConcurrencyPlus
// import LanguageServerProtocol
// import ProcessEnv
// import JSONRPC
//
// protocol LanguageExtension {
//     var client: RemoteLanguageServer { get }
//
// }
//
// final class UnrestrictedProcessTransport {
//     private var readHandler: ReadHandler = { _ in }
//     private let process: HostedProcess
//     private let taskQueue = TaskQueue()
//     private var subscription: AnyCancellable?
//     private let logger = Logger(subsystem: "com.chimehq.ChimeKit", category: "UnrestrictedProcessTransport")
//
//     init(process: HostedProcess) {
//         self.process = process
//     }
//
//     func beginMonitoringProcess() async throws {
//         let task = taskQueue.addOperation {
//             self.subscription = try await self.process.processEventPublisher
//                 .sink(receiveCompletion: { _ in
//                 }, receiveValue: { [weak self] event in
//                     switch event {
//                     case .stdout(let data):
//                         self?.readHandler(data)
//                     case .stderr(let data):
//                         let output = String(data: data, encoding: .utf8) ?? ""
//
//                         self?.logger.info("stderr: \(output, privacy: .public)")
//                     default:
//                         break
//                     }
//                 })
//         }
//
//         try await task.value
//     }
// }
//
// extension UnrestrictedProcessTransport: DataTransport {
//     func write(_ data: Data) {
//         taskQueue.addOperation {
//             try await self.process.write(data)
//         }
//     }
//
//     func setReaderHandler(_ handler: @escaping ReadHandler) {
//         self.readHandler = handler
//     }
//
//     func close() {
//         subscription?.cancel()
//     }
// }
//
// /// Provides an interface to a LSP language server hosted by an intermediary process.
// public class RemoteLanguageServer {
//     private let wrappedServer: JSONRPCLanguageServer
//     private var subscription: AnyCancellable?
//     private let logger = Logger(subsystem: "com.chimehq.ChimeKit", category: "RemoteLanguageServer")
//
//     private let process: HostedProcess
//     private let taskQueue = TaskQueue()
//     public var terminationHandler: (() -> Void)?
//
//     init(named serviceName: String, parameters: Process.ExecutionParameters) throws {
//         self.process = HostedProcess(named: serviceName, parameters: parameters)
//         let transport = UnrestrictedProcessTransport(process: process)
//         self.wrappedServer = JSONRPCLanguageServer(dataTransport: transport)
//
//         taskQueue.addOperation {
//             self.logger.debug("launching remote server")
//
//             do {
//                 try await self.process.launch()
//
//                 self.subscription = try await self.process.processEventPublisher
//                     .sink(receiveCompletion: { _ in
//
//                     }, receiveValue: { event in
//                         switch event {
//                         case .terminated:
//                             self.terminationHandler?()
//                         default:
//                             break
//                         }
//                     })
//
//                 try await transport.beginMonitoringProcess()
//             } catch {
//                 self.logger.error("failed to launch: \(String(describing: error), privacy: .public)")
//             }
//         }
//     }
//
//     private func stopProcess() {
//         self.taskQueue.addOperation {
//             do {
//                 try await self.process.terminate()
//             } catch {
//                 self.logger.error("failed to terminate: \(String(describing: error), privacy: .public)")
//             }
//         }
//     }
//
//     public var logMessages: Bool {
//         get { return wrappedServer.logMessages }
//         set { wrappedServer.logMessages = newValue }
//     }
// }
//
// extension RemoteLanguageServer: LanguageServerProtocol.Server {
//     public func setHandlers(_ handlers: ServerHandlers, completionHandler: @escaping (ServerError?) -> Void) {
//         wrappedServer.setHandlers(handlers, completionHandler: completionHandler)
//     }
//
//     public func sendNotification(_ notif: ClientNotification, completionHandler: @escaping (ServerError?) -> Void) {
//         taskQueue.addOperation {
//             self.wrappedServer.sendNotification(notif, completionHandler: completionHandler)
//         }
//     }
//
//     public func sendRequest<Response: Codable>(_ request: ClientRequest, completionHandler: @escaping (ServerResult<Response>) ->  Void) {
//         taskQueue.addOperation {
//             self.wrappedServer.sendRequest(request, completionHandler: { (result: ServerResult<Response>) in
//                 if case .success = result, case .shutdown = request {
//                     self.stopProcess()
//                 }
//
//                 completionHandler(result)
//             })
//         }
//     }
// }
```

**Review by activcoding — DISMISSED** [review]
Just pop in those spaces between slashes and code, and you're good to go 👍🏻

**Comment by Wouter01** [comment]
Instead of commenting out the code, I would just remove it alltogether, as it's not working, and probably out of date. The code is also from ChimeKit, and I should have referenced that.

**Comment by FastestMolasses** [comment]
Fixed!

**Review by Wouter01 — APPROVED** [review]

**Review by activcoding — APPROVED** [review]

**Pull request merged**